### PR TITLE
View list of comments

### PR DESCRIPTION
**Pull request by jakerbp**

### CODE REVIEW

File: components/CommentList.jsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { useEffect, useState } from "react";
+import axios from "axios";
+import CommentCard from "./CommentCard";
+
+const CommentList = ({ article_id }) => {
+  const [comments, setComments] = useState([]);
+  const [loadingComments, setLoadingComments] = useState(true);
+
+  useEffect(() => {
```
Comment (mezzDavies) [inline components/CommentList.jsx:9]: I would consider moving all API calls into a separate file and exporting them for use in you components - this makes you code a lot cleaner.

**Review by mezzDavies — CHANGES_REQUESTED** [review]
This is looking good but there is one thing that needs to be considered - what if a user choses to look at an article with no comments? There should be some feedback so that it doesn't look like the comments haven't loaded and that there actually aren't any. Please implement this.

File: src/utils.js [hunk 2]
```diff
@@ -26,4 +26,15 @@ const fetchArticle = ({ article_id }) => {
...
+  return newsApi
+    .get(`articles/${article_id}/comments`)
+    .then((res) => {
+      return res.data.articleComments;
+    })
+    .catch((err) => {
+      console.log(err);
+    });
+};
+
```
Comment (itskatherine) [inline src/utils.js:39]: Nice separation into this file 👍

File: components/CommentList.jsx (hunk 1 above)
Comment (itskatherine) [reply components/CommentList.jsx:9]: Smashed it 👍

File: components/CommentList.jsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    fetchComments({ article_id }).then((articleComments) => {
+      setComments(articleComments);
+      setLoadingComments(false);
+    });
+  }, []);
+
+  if (loadingComments) return <h2 className="loading">LOADING...</h2>;
+  if (!comments.length) return <h3 className="loading">No comments yet...</h3>;
```
Comment (itskatherine) [inline components/CommentList.jsx:17]: nice!

**Review by itskatherine — COMMENTED** [review]
Good to merge Jake 👍

**Pull request merged**